test/Microsoft.ML.CpuMath.UnitTests.netstandard/UnitTests.cs

```diff
@@ -0,0 +1,276 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Xunit;
+using Microsoft.ML.Runtime.Internal.CpuMath;
+
+namespace Microsoft.ML.CpuMath.UnitTests
+{
+    public class CpuMathUtilsUnitTests
+    {
+        private readonly float[][] testArrays;
+        private readonly int[] testIndexArray;
+        private const float DEFAULT_SCALE = 1.7f;
+        private FloatEqualityComparer comparer;
+
+        public CpuMathUtilsUnitTests()
+        {
+            // padded array whose length is a multiple of 4
+            float[] testArray1 = new float[8] { 1.96f, -2.38f, -9.76f, 13.84f, -106.37f, -26.93f, 32.45f, 3.29f };
+            // unpadded array whose length is not a multiple of 4.
+            float[] testArray2 = new float[7] { 1.96f, -2.38f, -9.76f, 13.84f, -106.37f, -26.93f, 32.45f };
+            testArrays = new float[][] { testArray1, testArray2 };
+            testIndexArray = new int[4] { 0, 2, 5, 6 };
+            comparer = new FloatEqualityComparer();
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        public unsafe void DotUTest(int test)
+        {
+            float[] src = (float[]) testArrays[test].Clone();
+            float[] dst = (float[]) src.Clone();
+            
+            for (int i = 0; i < dst.Length; i++)
+            {
+                dst[i] += 1;
+            }
+
+            var managedOutput = CpuMathUtils.DotProductDense(src, dst, dst.Length);
+
+            if (test == 0)
```

Comment:
Typically, this is handled by passing the "expected" value in as InlineData as well:

```C#
[InlineData(0, 13306.0376f)]
[InlineData(1, (13291.9235f)]
public unsafe void DotUTest(int test, float expected)
```